src/launchpad/artifacts/apple/zipped_xcarchive.py

```diff
@@ -494,27 +495,75 @@ def _parse_asset_element(self, item: dict[str, Any], parent_path: Path) -> Asset
             colorspace=colorspace,
         )
 
-    def _extract_binary_uuid(self, binary_path: Path) -> str | None:
-        try:
-            with open(binary_path, "rb") as f:
-                fat_binary: lief.MachO.FatBinary | None = lief.MachO.parse(f)  # type: ignore
+    def _parse_and_cache_all_binaries(self, binary_paths: List[Path]) -> None:
+        """Parse all binaries once, extracting UUIDs and caching LIEF objects for those with dSYMs."""
+        if self._dsym_info is None:
+            self._find_dsym_files()
+
+        for binary_path in binary_paths:
+            if not binary_path.exists():
+                logger.warning(f"Binary path does not exist: {binary_path}")
+                continue
+
+            try:
+                with open(binary_path, "rb") as f:
+                    fat_binary: lief.MachO.FatBinary | None = lief.MachO.parse(f)  # type: ignore
+
                 if fat_binary is None or fat_binary.size == 0:
                     logger.debug(f"Failed to parse binary with LIEF: {binary_path}")
-                    return None
+                    continue
 
                 binary = fat_binary.at(0)
 
-                # Look for UUID load command
+                extracted_uuid = None
                 for command in binary.commands:
                     if command.command == lief.MachO.LoadCommand.TYPE.UUID:
                         if isinstance(command, lief.MachO.UUIDCommand):
                             uuid_bytes = bytes(command.uuid)
                             uuid_obj = uuid.UUID(bytes=uuid_bytes)
-                            return str(uuid_obj).upper()
+                            extracted_uuid = str(uuid_obj).upper()
+                            break
+
+                if extracted_uuid is None:
+                    logger.debug(f"No UUID command found in binary: {binary_path}")
+                    continue
+
+                self._binary_uuid_cache[binary_path] = extracted_uuid
+                if extracted_uuid in self._dsym_info:
```

Comment:
Same with this check, not entirely sure what it's doing. We should cache the lief.parse() call in every case unless I'm misunderstanding something.